src/java/com/puppetlabs/jruby_utils/pool/JRubyPool.java

```diff
@@ -192,15 +204,11 @@ public E borrowItemWithTimeout(long timeout, TimeUnit unit) throws
 
     @Override
     public void releaseItem(E e) {
-        releaseItem(e, true);
-    }
-
-    @Override
-    public void releaseItem(E e, boolean returnToPool) {
         final ReentrantLock lock = this.queueLock;
         lock.lock();
         try {
-            if (returnToPool) {
+            // Do nothing if the item being returned is the poisonPill
```

Comment:
We could ignore any subsequent calls to `insertPill` or log a warning. If a pill has been inserted then the server should be shutting down. We could probably ignore it, since the error that caused a pill to be inserted should be thrown or logged where it was caught. It's not the pool's job to handle that.

I think the exception held in the pill can be returned to the user through an http request though? Not sure if that changes things.

@camlow325 thoughts?